ui/src/app/views/concept-set-details/component.tsx

```diff
@@ -0,0 +1,321 @@
+      <TooltipTrigger content={<div>Requires Write Permission</div>}
+                      disabled={canEdit}>
+        <MenuItem icon='pencil'
+                  onClick={() => onEdit}
+                  disabled={!canEdit}>
+          Edit
+        </MenuItem>
+      </TooltipTrigger>
+      <TooltipTrigger content={<div>Requires Owner Permission</div>}
+                      disabled={canDelete}>
+        <MenuItem icon='trash' onClick={onDelete} disabled={!canDelete}>
+          Delete
+        </MenuItem>
+      </TooltipTrigger>
+    </React.Fragment>}
+  >
+    <Clickable disabled={disabled} data-test-id='workspace-menu'>
+      <ClrIcon shape='ellipsis-vertical' size={21}
+               style={{color: disabled ? '#9B9B9B' : '#216FB4', marginLeft: -9,
+                 cursor: disabled ? 'auto' : 'pointer'}}/>
+    </Clickable>
+  </PopupTrigger>;
+};
+
+export const ConceptSetDetails =
+  fp.flow(withUrlParams(), withCurrentWorkspace())(class extends React.Component<{
+    urlParams: any, workspace: WorkspaceData}, {
+      conceptSet: ConceptSet, deleting: boolean, deletingConcepts: boolean,
+      deleteSubmitting: boolean, editName: string, editDescription: string, editSaving: boolean,
+      error: boolean, errorMessage: string, editing: boolean, loading: boolean,
+      selectedConcepts: Concept[]}> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        conceptSet: undefined,
+        editName: '',
+        editDescription: '',
+        editSaving: false,
+        error: false,
+        errorMessage: '',
+        deleting: false,
+        deletingConcepts: false,
+        deleteSubmitting: false,
+        editing: false,
+        loading: true,
+        selectedConcepts: []
+      };
+    }
+
+    componentDidMount() {
+      this.getConceptSet();
+    }
+
+    async getConceptSet() {
+      const {urlParams: {ns, wsid, csid}} = this.props;
+      try {
+        const resp = await conceptSetsApi().getConceptSet(ns, wsid, csid);
+        this.setState({conceptSet: resp, editName: resp.name,
+          editDescription: resp.description, loading: false});
+        currentConceptSetStore.next(resp);
+      } catch (error) {
+        console.log(error);
+        // TODO: what do we do with resources not found?  Currently we just have an endless spinner
+        // Maybe want to think about designing an AoU not found page for better UX
+      }
+    }
+
+    async submitEdits() {
+      const {urlParams: {ns, wsid, csid}} = this.props;
+      const {conceptSet, editName, editDescription} = this.state;
+      try {
+        this.setState({editSaving: true});
+        await conceptSetsApi().updateConceptSet(ns, wsid, csid,
+          {...conceptSet, name: editName, description: editDescription});
+        await this.getConceptSet();
+        this.setState({editing: false, editSaving: false});
+      } catch (error) {
+        console.log(error);
+        this.setState({editing: false, editSaving: false});
```

Comment:
This feels a little odd -- if an error occurs, we do the same thing as if it succeeds? I'd almost rather not change state, and leave the same TODO here as in L118, e.g. we should come back to this & clean up our error-handling strategy en masse.

Alternatively: could you adopt the same approach as L152 over here?

Reply:
yes i think the state-setting should be in a `finally` block